storage/umass/umass.c

```diff
@@ -802,81 +802,96 @@ static int umass_mountRoot(umass_dev_t *dev)
 }
 
 
-static int umass_handleInsertion(usb_driver_t *drv, usb_devinfo_t *insertion)
+static int umass_handleInsertion(usb_driver_t *drv, usb_devinfo_t *insertion, usb_event_insertion_t *event)
 {
 	int err;
 	umass_dev_t *dev;
 	oid_t oid;
-	int ret;
-
-	fprintf(stderr, "umass: pending insertion\n");
 
 	mutexLock(umass_common.lock);
 
-	if ((dev = umass_devAlloc()) == NULL) {
-		fprintf(stderr, "umass: devAlloc failed\n");
-		return -ENOMEM;
-	}
+	do {
+		dev = _umass_devAlloc();
+		if (dev == NULL) {
+			fprintf(stderr, "umass: devAlloc failed\n");
+			err = -ENOMEM;
+			break;
+		}
 
-	dev->drv = drv;
-	dev->instance = *insertion;
-	dev->pipeCtrl = usb_open(drv, insertion, usb_transfer_control, 0);
-	if (dev->pipeCtrl < 0) {
-		free(dev);
-		fprintf(stderr, "umass: usb_open failed\n");
-		return -EINVAL;
-	}
+		dev->drv = drv;
+		dev->instance = *insertion;
+		dev->pipeCtrl = usb_open(drv, insertion, usb_transfer_control, 0);
+		if (dev->pipeCtrl < 0) {
+			fprintf(stderr, "umass: usb_open failed\n");
+			free(dev->lock);
```

Comment:
```suggestion
			resourceDestroy(dev->lock);
```

Applies to many places below.

Maybe create `_umass_devFree()`? Do we need to free/remove a node allocated/inserted via `idtree_alloc()` in `_umass_devAlloc()`?